lib/console/teams/team.ex

```diff
@@ -6,15 +6,21 @@ defmodule Console.Teams.Team do
   @foreign_key_type :binary_id
   schema "teams" do
     field :name, :string
-    has_many :memberships, Console.Teams.Membership
+    field :encryption_version, :binary
+    field :private_key, Cloak.EncryptedBinaryField
+    field :public_key, Cloak.EncryptedBinaryField
+    field :address, :binary
+    field :address_b58, :string
+
+    has_many :memberships, Console.Teams.Membership, on_delete: :delete_all
     many_to_many :users, Console.Auth.User, join_through: "memberships"
-    has_many :invitations, Console.Teams.Invitation
+    has_many :invitations, Console.Teams.Invitation, on_delete: :delete_all
 
-    has_many :devices, Console.Devices.Device
-    has_many :gateways, Console.Gateways.Gateway
-    has_many :channels, Console.Channels.Channel
-    has_many :audit_trails, Console.AuditTrails.AuditTrail
-    has_many :notifications, Console.Notifications.Notification
+    has_many :devices, Console.Devices.Device, on_delete: :delete_all
+    has_many :gateways, Console.Gateways.Gateway, on_delete: :delete_all
```

Comment:
does this mean the gateways get deleted if the team gets deleted? isn't that bad because a gateway could be owned by someone not in the team?

Reply:
in this case, we're not exposing team deletion to users yet. This was just so I could clean up my teams locally -- I had too many from testing.

but also, teams own gateways, users don't. that's why the team gets the private key. any actions team members take are on behalf of the team